samtranslator/model/api/http_api_generator.py

```diff
@@ -164,6 +174,137 @@ def _add_cors(self):
         # Assign the OpenApi back to template
         self.definition_body = editor.openapi
 
+    def _construct_api_domain(self, http_api):
```

Comment:
This method is about 100 lines long, and is conceptually difficult to grasp. Could you break the components apart? It looks like this could be split into Domain, BasePath, and RouteSettings methods.